### PR TITLE
Subsets creation

### PR DESCRIPTION
**Pull request by kaszycaO**
Whats new?
+ add main dataset sampling
+ add new parametr `subset_ratio`

Whats now?
+ we can run experiments on a sample of dataset

### CODE REVIEW

File: personalized_active_learning/datasets/base.py [hunk 1]
```diff
@@ -205,14 +210,25 @@ def setup(self, stage: Optional[str] = None) -> None:
             stage: Left for compatibility with pytorch lighting.
 
         Prepare data to use. Currently this method:
-            1. Splits data into training, validation, test sets.
-            2. Limits annotations if needed.
-            3. Filters annotations if needed.
-            4. Computes annotators biases.
-            5. Applies major voting mechanism if needed.
-            6. Finetunes text embeddings if needed.
+            1. Create subset of data if needed
```
Comment (cicheck) [inline personalized_active_learning/datasets/base.py:213]: Kudos for keeping docstring up to date!

File: personalized_active_learning/datasets/base.py [hunk 2]
```diff
@@ -404,22 +463,8 @@ def _split_data(self) -> None:
...
-            if "split" not in self.annotations.columns:
-                raise Exception(
-                    "Split mode {0} is used but no column split in {1}".format(
-                        self.split_mode.value,
-                        self.annotations.columns,
-                    )
-                )
             return
         elif self.split_mode == SplitMode.TEXTS:
-            if "fold" not in self.annotations.columns:
```
Comment (cicheck) [inline personalized_active_learning/datasets/base.py:416]: Why it is removed?

Comment (kaszycaO) [reply personalized_active_learning/datasets/base.py:416]: Cause the check is done in line 273. I think that double check is not necessary

**Review by mateuszkochanek — APPROVED** [review]
lgtm

**Pull request merged**